data/lib/core/creature.lua

```diff
@@ -50,3 +50,48 @@ end
 function Creature.isTile(self)
 	return false
 end
+
+function Creature:setMonsterOutfit(monster, time)
+	if not self then
+		return false
+	end
+
+	local monsterType = MonsterType(monster)
+	if not monsterType then
+		return false
+	end
+
+	if self:getPlayer() and not getPlayerFlagValue(self, PlayerFlag_CanIllusionAll) then
+		if not monsterType:isIllusionable() then
```

Comment:
You can merge those checks: `if self:getPlayer() and not getPlayerFlagValue(self, PlayerFlag_CanIllusionAll) and not monsterType:isIllusionable() then` or even `if self:getPlayer() and not (getPlayerFlagValue(self, PlayerFlag_CanIllusionAll) or monsterType:isIllusionable()) then`